### PR TITLE
New circular shape feature 

### PR DESCRIPTION
**Pull request by masmariuss**
## Description

Created _buildFlagWidget(), _buildCircularFlag() and _buildRectangularFlag() methods for displaying circular or rectangle flag shape.
Reformulated README.md

## Type of Change

- ✨ New feature (non-breaking change which adds functionality)
- 📝 Documentation

### CODE REVIEW

**Comment by ahmetakil** [comment]
Can you instead make this take an enum not a String so it's less error prone

**Comment by masmariuss** [comment]
> Can you instead make this take an enum not a String so it's less error prone

Thanks for suggest that change, I'm going to implement it. 
For sure, this change will make code less error-prone.

File: lib/src/country_flags.dart [hunk 1]
```diff
@@ -22,13 +23,15 @@ class CountryFlag extends StatelessWidget {
...
           borderRadius: borderRadius,
+          shape: shape,
         );
 
   /// Create an instance of [CountryFlag] based on a country code.
   ///
   /// {@macro country_flags}
   CountryFlag.fromCountryCode(
     String countryCode, {
+    required Shape shape,
```
Comment (arturograu) [inline lib/src/country_flags.dart:34]: Here we can better define a default value with the `Shape.rectangle`.

File: lib/src/country_flags.dart [hunk 2]
```diff
@@ -92,4 +94,47 @@ class CountryFlag extends StatelessWidget {
       ),
     );
   }
+
+  /// Build the flag widget according to the specified shape.
+  Widget _buildFlagWidget() {
+    switch (shape) {
```
Comment (arturograu) [inline lib/src/country_flags.dart:100]: This is a good opportunity to get advantage of the new pattern matching feature introduced in Dart 3.0:
```
switch (shape) {
   Shape.circle => const _buildCircularFlag(),
   Shape.rectangle => const _buildRectangularFlag(),
},
```

File: lib/src/country_flags.dart [hunk 3]
```diff
@@ -92,4 +94,47 @@ class CountryFlag extends StatelessWidget {
...
+        return _buildCircularFlag();
+      case Shape.rectangle:
+        return _buildRectangularFlag();
+    }
+  }
+
+  /// Build the circular flag widget.
+  Widget _buildCircularFlag() {
+    return ClipOval(
+      child: ScalableImageWidget.fromSISource(
```
Comment (arturograu) [inline lib/src/country_flags.dart:111]: Let's transform this `ScalableImageWidget.fromSISource` into a custom widget so we can reuse it in the `_buildCircularFlag()` method and the `_buildRectangularFlag()`.

**Review by arturograu — APPROVED** [review]

**Pull request merged**